include/biscuit/cpuinfo.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c), 2022, KNS Group LLC (YADRO)
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#pragma once
+
+#include <biscuit/assembler.hpp>
+#include <biscuit/registers.hpp>
+#include <cstddef>
+#include <cstdint>
+
+#include <sys/auxv.h>
```

Comment:
This cpuinfo file should currently only be compiled on unix-only environments or have its unix implementation and includes hidden behind an ifdef.

Since this file is unconditionally compiled, this will break building on windows

Reply:
Fixed, thank you. I've hidden unix-specific implementation details behind ifdefs. 

cpuinfo.exe output on Windows x64:
```
$ ./cpuinfo.exe
Has I:0
Has M:0
Has A:0
Has F:0
Has D:0
Has C:0
Has V:0
```

compiled with
```
cmake -D BISCUIT_CODE_BUFFER_MMAP=OFF \
      -D BUILD_TESTING=ON \
      -D BUILD_EXAMPLES=ON \
      ..
      
 cmake --build .
```